components/CreateCycle/CreateCycle.jsx

```diff
@@ -50,20 +53,22 @@ export default ({ navigation }) => {
 
   const createNewCycle = async (newCycle) => {
     const currentUser = firebase.auth().currentUser.uid;
-
     const cycleRef = firebase.firestore()
       .collection('users')
       .doc(currentUser)
       .collection('cycles');
-
-    const cycleDoc = await cycleRef.add(newCycle);
+    let cycleDoc;
+    if (isNewCycle) {
+      cycleDoc = await cycleRef.add(newCycle);
+    } else {
+      cycleDoc = await cycleRef.doc(cycleID).update(newCycle);
+    }
 
     dispatch(actions.cycles.addCycle({
       workouts: newCycle.workoutIds,
       name: newCycle.name,
       id: cycleDoc.id,
     }));
```

Comment:
Does this code add a new cycle to redux even when editing?

Reply:
This actually breaks when you edit a workout. It just shows as a warning bc its a promise rejection but when I edit a cycle I get `[Unhandled promise rejection: TypeError: undefined is not an object (evaluating 'cycleDoc.id')]` I think you can fix this by creating a new action/reducer for specifically updating a cycle and call that when editing instead of addCycle